### DISCUSSION TITLE
In-process code a.k.a. stored procedures

### DISCUSSION THREAD

**Discussion by tslocke** [geldata/gel#2687] [Q&A]
Some "next gen" databases, like Datomic and ArangoDB make a big deal out of the fact they can run application code in the same process as the database itself for wildly improved performance in certain situations. Of course this is just the old idea of stored procedures, but potentially using the same language as the app (Clojure, JS...)

Uses could include:

 - complex retrieval of aggregated data based on logic not expressible in the query language
 - row level authorisation or data validation using complex logic (not expressible...)
 - updates that maintain complex data integrity rules

Taken to extreme, you can ultimately get to "the database IS the backend", which is pretty sweet.

I'm wondering what the EdgeDB story is here. Something like this exists already? On the roadmap? Query language is powerful enough that it's not needed? Not gonna happen?

Thanks

Tom

**Comment by elprans** [comment]
> you can ultimately get to "the database IS the backend"

Yes! This is one of the big driving factors in the design of EdgeDB.  We are actively exploring the extension model, where you'd be able to bring your own language and plug code either as middleware between the frontend and the database or plain old stored procedures.  The current plan is to attempt to leverage the WebAssembly ecosystem.
